integrations/gcp/main.py

```diff
@@ -0,0 +1,76 @@
+import http
+
+from fastapi import Response
+from loguru import logger
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+from starlette.requests import Request
+
+from gcp_core.utils import (
+    GotFeedCreatedSuccessfullyMessage,
+    create_feed_from_ocean_config,
+    create_gcp_client_from_ocean_config,
+    parse_subscription_message_from_request,
+)
+
+
+@ocean.on_start()
+async def setup_feed() -> None:
+    gcp_client = create_gcp_client_from_ocean_config()
+    try:
+        feed = create_feed_from_ocean_config()
+        await gcp_client.create_assets_feed_if_not_exists(feed)
+    except Exception as e:
+        logger.warning(
+            f"Couldn't setup feed, continuing integration without real-time events. Error: {str(e)}"
+        )
+
+
+@ocean.on_resync()
+async def resync_resources(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    # List of supported assets: https://cloud.google.com/asset-inventory/docs/supported-asset-types
+    gcp_client = create_gcp_client_from_ocean_config()
+    async for resources in gcp_client.generate_resources(kind):
+        yield resources
```

Comment:
didnt we agree on specific implementations in addition to this?

Reply:
The only problematic kind is Topic, should I add a specific implementation for it?